pkg/core/client.go

```diff
@@ -38,7 +38,7 @@ type Client interface {
 
 	ListChannels(options ListChannelOptions) (*eventing.ChannelList, error)
 	CreateChannel(options CreateChannelOptions) (*eventing.Channel, error)
-	DeleteChannel(options DeleteChannelOptions) error
+	DeleteChannel(options DeleteChannelOptions) []error
```

Comment:
consider:

- pluralize channel
- merge errors before returning

```
DeleteChannels(options DeleteChannelOptions) error
```


We should also seriously consider leaving this interface unchanged and instead invoke it multiple times for a batch delete. That would give the caller more control over concurrency and retry.

Reply:
👍

Reply:
Thinking about it, I guess the last option you mention is probably the best.